src/components/App.tsx

```diff
@@ -244,17 +244,17 @@ export class App extends React.Component<any, AppState> {
     }
   };
 
-  private initializeSocketClient = () => {
+  private initializeSocketClient = (
+    opts: { showLoadingState: boolean } = {
+      showLoadingState: true,
+    },
+  ) => {
     if (this.socket) {
       return;
     }
     const roomMatch = getCollaborationLinkData(window.location.href);
     if (roomMatch) {
-      let isInitialized = false;
-      this.setState({
-        isCollaborating: true,
-        isLoading: true,
-      });
```

Comment:
I've moved this at the end of the callback because previously this caused initialization 2x due to `this.socket` not being set by the the `componentDidUpdate` was called.